application/src/main/java/org/togetherjava/tjbot/commands/tag/TagManageCommand.java

```
@@ -0,0 +1,233 @@
+                    Button.of(ButtonStyle.DANGER, generateComponentId(member.getId(), tagId),
+                            "Of course!"),
+                    Button.of(ButtonStyle.SUCCESS, generateComponentId(member.getId()), "Abort"))
+            .queue();
+    }
+
+    private void createTag(SlashCommandEvent event) {
+        String tagId = event.getOption("id").getAsString();
+        String content = event.getOption(CONTENT).getAsString();
+
+        if (tagSystem.exists(tagId)) {
+            event.reply("This tag already exists").setEphemeral(true).queue();
+
+            return;
+        }
+
+        tagSystem.put(tagId, content);
+
+        event
+            .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                .setTitle(SUCCESS)
+                .setTimestamp(LocalDateTime.now())
+                .setFooter(event.getUser().getAsTag())
+                .setDescription("Successfully created tag '" + tagId + "'")
+                .build())
+            .queue();
+    }
+
+    private void createIdTag(SlashCommandEvent event) {
+        String tagId = event.getOption(TAG_MINUS_ID).getAsString();
+        long messageId = event.getOption(MESSAGE_MINUS_ID).getAsLong();
+
+        if (tagSystem.exists(tagId)) {
+            event.reply("This tag already exists").setEphemeral(true).queue();
+
+            return;
+        }
+
+        event.getMessageChannel().retrieveMessageById(messageId).queue(message -> {
+            tagSystem.put(tagId, message.getContentRaw());
+
+            event
+                .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                    .setTitle(SUCCESS)
+                    .setTimestamp(LocalDateTime.now())
+                    .setFooter(event.getUser().getAsTag())
+                    .setDescription("Successfully created tag '" + tagId + "'")
+                    .build())
+                .queue();
+        }, failure -> {
+            if (failure instanceof ErrorResponseException ex
```

swallows other errors. check the previous message and apply the improvements here as well.

---

Yea, I forgot to do this here too.